index.js

```diff
@@ -88,15 +88,15 @@ class ApiFactory {
 
 			let proxyPath = '/{proxy+}';
 			// default to defined path when proxy is not specified.
-			if (event.resource !== proxyPath) {
+			if (!event.resource.includes(proxyPath)) {
 				if (mainEventHandler && mainEventHandler[event.resource]) {
 					definedRoute = mainEventHandler[event.resource];
 				} else if (anyEventHandler && anyEventHandler[event.resource]) {
 					definedRoute = anyEventHandler[event.resource];
 				}
 			} else {
 				// if it is a proxy path then then look up the proxied value.
-				let map = mapExapander.getMapValue(apiFactory.ProxyRoutes[event.httpMethod], event.pathParameters.proxy);
+				let map = mapExapander.getMapValue(apiFactory.ProxyRoutes[event.httpMethod], event.path);
 				if (map) {
 					definedRoute = map.value;
 					event.path = event.pathParameters.proxy;
```

Comment:
So I've deployed with these changes to a test stage. My path is https://domain.name/test/v1/path. The code seems to be working, I'm able to hit all my endpoints and the results are as expected.

Reply:
So I've made a deployment with these changes to a test stage and it seems to be working fine. 
My path is https://domain.name/test/v1/path. 

In my open-api factory I am configuring my routes to be /v1/path

and my Serverless config has this path `path: /v1/{proxy+}`

Reply:
I can try to add some tests that you mentioned

Reply:
Also I think the most important line is 101 ? We want the definedRoute to be set so we can get the handler.

Reply:
(102, 103) my bad, yes we need the `definedRoute`, I meant the handling include the `/path` part.

> So I've made a deployment with these changes to a test stage and it seems to be working fine.
> My path is https://domain.name/test/v1/path.

Can I ask what your custom domain configuration is, do you have the base path set to be `/` or `/test`

Reply:
So this is what Serverless generated for cloud formation, /test came from the stage 

        ```
  "ServiceEndpoint": {
            "Description": "URL of the service endpoint",
            "Value": {
              "Fn::Join": [
                "",
                [
                  "https://",
                  {
                    "Ref": "ApiGatewayRestApi"
                  },
                  ".execute-api.us-east-1.",
                  {
                    "Ref": "AWS::URLSuffix"
                  },
                  "/test"
                ]
              ]
            }
          }
```

Reply:
According to that, it seems the answer is `yes` which means that the `/test` will *not* show up in the `event.path`.  We can confirm this by going to [API Gateway](https://eu-west-1.console.aws.amazon.com/apigateway/home?region=eu-west-1#/custom-domain-names) and clicking on `Custom Domain Names` => `show base path mappings`.  Which would show something like:
![image](https://user-images.githubusercontent.com/5056218/50488177-73780000-0a01-11e9-81a1-1fae9eaa06d9.png)

Instead of this:
![image](https://user-images.githubusercontent.com/5056218/50488183-796de100-0a01-11e9-8d83-f8eba1d93505.png)

In the case of using Serverless, or the first example the code does work correctly, but in the case of the second one, it does not.  I hope it is clear why that might be, I'm happy to explain it further. In any case we should assume that the `path` field can be contaminated with this value.  I've suggested one possible solution, another one would be remove the `stage` using the api gateway field found in the `requestContext`, but I thought using the `event.resource` was cleaner.

Reply:
Added the requested changes let me know if the mapExpander test I wrote is what you expected